subscription.go

```diff
@@ -157,16 +157,23 @@ func (r *marathonClient) registerSSESubscription() error {
 	if r.subscribedToSSE {
 		return nil
 	}
-
 	// Get a member from the cluster
 	marathon, err := r.cluster.GetMember()
 	if err != nil {
 		return err
 	}
 	url := fmt.Sprintf("%s/%s", marathon, marathonAPIEventStream)
-
-	// Try to connect to stream
-	stream, err := eventsource.Subscribe(url, "")
+	// step: create a new request for the stream
+	request, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return err
+	}
+	// step: add any basic authentication
+	if r.config.HTTPBasicAuthUser != "" && r.config.HTTPBasicPassword != "" {
+		request.SetBasicAuth(r.config.HTTPBasicAuthUser, r.config.HTTPBasicPassword)
+	}
```

Comment:
We already run a similar, yet already slightly deviating logic in [apiCall()](https://github.com/gambol99/go-marathon/blob/fbaf44af7084997a4bb1fd1749a271640d17f671/client.go#L247). What would you think about extracting it into a central utility function that we could use from both call sites?